saiph/reduction/utils/svd.py

```diff
@@ -1,26 +1,121 @@
-from typing import Any, Tuple
+from typing import Any, Optional, Tuple
 
+import numpy as np
 import pandas as pd
 from numpy.typing import NDArray
 from scipy import linalg
 from sklearn.utils import extmath
 
 
-def SVD(
-    df: pd.DataFrame, svd_flip: bool = True
+def get_svd(
+    df: pd.DataFrame,
+    nf: Optional[int] = None,
```

Comment:
do we need an arg to either `use_full_svd`, `use_randomized_svd` or `use_auto_svd` ?

Reply:
We have to think about it 👍